File: src/resources/ConnectionTestResults.ts
export default interface ConnectionTestResults {
  transport: string;
  audit_connection_result: ConnectionSettingsTestResult;
  monitoring_connection_result: ConnectionSettingsTestResult;
  broker_connection_result: ConnectionSettingsTestResult;
}

export interface ConnectionSettingsTestResult {
  connection_successful: boolean;
  connection_error_messages: string[];
  diagnostics: string;
}

File: src/Particular.LicensingComponent.Contracts/ConnectionTestResults.cs
namespace Particular.LicensingComponent.Contracts;

public class ConnectionTestResults(
    string transport,
    ConnectionSettingsTestResult auditConnectionResult,
    ConnectionSettingsTestResult monitoringConnectionResult,
    ConnectionSettingsTestResult brokerConnectionResult)
{
    public string Transport { get; } = transport;

    public ConnectionSettingsTestResult AuditConnectionResult { get; } = auditConnectionResult;
    public ConnectionSettingsTestResult MonitoringConnectionResult { get; } = monitoringConnectionResult;
    public ConnectionSettingsTestResult BrokerConnectionResult { get; } = brokerConnectionResult;
}